### PR TITLE
OF-2440: Increase default cache sizes

### PR DESCRIPTION
**Pull request by guusdk**
Most of the default cache sizes were defined a long time ago, when RAM was expensive.

Given that most servers nowadays have plenty of RAM, increasing the default sizes will benefit those instances that see heavy usage, at nearly no expense.

### CODE REVIEW

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 1]
```diff
@@ -220,17 +220,17 @@ public class CacheFactory {
...
         cacheProps.put(PROPERTY_PREFIX_CACHE + "entityCapabilities" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofDays(2).toMillis());
         cacheProps.put(PROPERTY_PREFIX_CACHE + "entityCapabilitiesUsers" + PROPERTY_SUFFIX_SIZE, -1L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "entityCapabilitiesUsers" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofDays(2).toMillis());
         cacheProps.put(PROPERTY_PREFIX_CACHE + "pluginCacheInfo" + PROPERTY_SUFFIX_SIZE, -1L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "pluginCacheInfo" + PROPERTY_SUFFIX_MAX_LIFE_TIME, -1L);
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "pepServiceManager" + PROPERTY_SUFFIX_SIZE, 1024L * 1024 * 10);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "pepServiceManager" + PROPERTY_SUFFIX_SIZE, 10 * 1024L * 1024 * 10);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "pepServiceManager" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(30).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "publishedItems" + PROPERTY_SUFFIX_SIZE, 1024L * 1024 * 10);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "publishedItems" + PROPERTY_SUFFIX_SIZE, 10 * 1024L * 1024 * 10);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:233]: Some of these look to have had tweaks in the past, and so now look weirdly unreadable.

Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java]: I've force-pushed minor changes to account for these. Any others that stand out to you?

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 2]
```diff
@@ -150,44 +150,44 @@ public class CacheFactory {
...
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "fileTransfer" + PROPERTY_SUFFIX_SIZE, 10 * 128 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "fileTransfer" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(10).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "multicast" + PROPERTY_SUFFIX_SIZE, 128 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "multicast" + PROPERTY_SUFFIX_SIZE, 10 * 128 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "multicast" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofDays(1).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "offlinemessage" + PROPERTY_SUFFIX_SIZE, 100 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "offlinemessage" + PROPERTY_SUFFIX_SIZE, 1024 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "offlinemessage" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofHours(12).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "pop3" + PROPERTY_SUFFIX_SIZE, 512 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "pop3" + PROPERTY_SUFFIX_SIZE, 10 * 512 * 1024L);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:161]: What's this one saying?
10 lots of 512k?
Does 5 lots of 1024k (so 5 lots of 1M) conveys the size more easily, but does it lose some essence that I can't see?

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 3]
```diff
@@ -150,44 +150,44 @@ public class CacheFactory {
...
         cacheProps.put(PROPERTY_PREFIX_CACHE + "transferProxy" + PROPERTY_SUFFIX_SIZE, -1L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "transferProxy" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(1).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "group" + PROPERTY_SUFFIX_SIZE, 1024 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "group" + PROPERTY_SUFFIX_SIZE, 10 * 1024 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "group" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(15).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "lockOutCache" + PROPERTY_SUFFIX_SIZE, 1024 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "lockOutCache" + PROPERTY_SUFFIX_SIZE, 10 * 1024 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "lockOutCache" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(15).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "groupMeta" + PROPERTY_SUFFIX_SIZE, 512 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "groupMeta" + PROPERTY_SUFFIX_SIZE, 10 * 512 * 1024L);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:169]: Ditto
Is 512 relevant, or should this be in 1024 units?

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 4]
```diff
@@ -150,44 +150,44 @@ public class CacheFactory {
...
         cacheProps.put(PROPERTY_PREFIX_CACHE + "groupMeta" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(15).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "groupSharingMeta" + PROPERTY_SUFFIX_SIZE, 512 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "groupSharingMeta" + PROPERTY_SUFFIX_SIZE, 10 * 512 * 1024L);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:171]: Ditto

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 5]
```diff
@@ -150,44 +150,44 @@ public class CacheFactory {
...
         cacheProps.put(PROPERTY_PREFIX_CACHE + "groupSharingMeta" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(15).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "username2roster" + PROPERTY_SUFFIX_SIZE, 1024 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "username2roster" + PROPERTY_SUFFIX_SIZE, 10 * 1024 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "username2roster" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(30).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "username2rosterItems" + PROPERTY_SUFFIX_SIZE, 1024 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "username2rosterItems" + PROPERTY_SUFFIX_SIZE, 10 * 1024 * 1024L);
         cacheProps.put(PROPERTY_PREFIX_CACHE + "username2rosterItems" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofMinutes(10).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "javascript" + PROPERTY_SUFFIX_SIZE, 128 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "javascript" + PROPERTY_SUFFIX_SIZE, 10 * 128 * 1024L);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:177]: Ditto, but with 128 as the maybe relevant number

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java [hunk 6]
```diff
@@ -150,44 +150,44 @@ public class CacheFactory {
...
         cacheProps.put(PROPERTY_PREFIX_CACHE + "javascript" + PROPERTY_SUFFIX_MAX_LIFE_TIME, Duration.ofDays(10).toMillis());
-        cacheProps.put(PROPERTY_PREFIX_CACHE + "ldap" + PROPERTY_SUFFIX_SIZE, 512 * 1024L);
+        cacheProps.put(PROPERTY_PREFIX_CACHE + "ldap" + PROPERTY_SUFFIX_SIZE, 10 * 512 * 1024L);
```
Comment (Fishbowler) [inline xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java:179]: Ditto again for 512

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java (hunk 1 above)
Comment (Fishbowler) [reply xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java]: I've added comments on the first 5 I found.

I don't know if the numbers infer relevance (128 or 512, for example, might have relevance to JID sizes or something)

Or perhaps the `10 *` prefix gives good history of the size?

If not, this are still nitpicks on readability and could still be ignored :)

Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java]: What about this? (force-pushed again)

File: xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java (hunk 2 above)
Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/util/cache/CacheFactory.java]: I don't think it's saying much. I've replaced it with a single long, formatted for readability.

**Review by Fishbowler — APPROVED** [review]

**Pull request merged**